### DISCUSSION TITLE
stale connections are not being closed even after they exceed the IdleTimeout period

### DISCUSSION THREAD

**Discussion by prateeeka-nadimpalli** [redis/go-redis#3112] [General]
i have my redis client configuration as below

```
PoolSize:     500,
PoolTimeout:  30 * time.Second,
IdleTimeout:  2 * time.Minute,
MinIdleConns: 100,
MaxRetries:   3,
```

when iam taking the redis connection statistics of my app, i observed that the stale connections keeps on increasing . It started with 10 and keep on increasing to 7000. I am taking these these stats for every minute.

```
Time: 2024-09-11T08:57:52Z, Total Conns: 500, Idle Conns: 500, Stale Conns: 6558, Hits: 583330, Misses: 37959, Timeout: 0
Time: 2024-09-11T08:58:52Z, Total Conns: 100, Idle Conns: 100, Stale Conns: 7013, Hits: 583330, Misses: 37959, Timeout: 0
Time: 2024-09-11T08:59:52Z, Total Conns: 100, Idle Conns: 100, Stale Conns: 7058, Hits: 583330, Misses: 37959, Timeout: 0
Time: 2024-09-11T09:00:52Z, Total Conns: 500, Idle Conns: 2, Stale Conns: 7058, Hits: 600387, Misses: 37979, Timeout: 0
Time: 2024-09-11T09:01:54Z, Total Conns: 500, Idle Conns: 500, Stale Conns: 7058, Hits: 625269, Misses: 37979, Timeout: 0
Time: 2024-09-11T09:02:54Z, Total Conns: 500, Idle Conns: 500, Stale Conns: 7058, Hits: 625269, Misses: 37979, Timeout: 0
Time: 2024-09-11T09:03:54Z, Total Conns: 100, Idle Conns: 100, Stale Conns: 7463, Hits: 625318, Misses: 37979, Timeout: 0
Time: 2024-09-11T09:04:54Z, Total Conns: 100, Idle Conns: 100, Stale Conns: 7556, Hits: 625318, Misses: 37979, Timeout: 0
Time: 2024-09-11T09:05:54Z, Total Conns: 500, Idle Conns: 8, Stale Conns: 7556, Hits: 643947, Misses: 37998, Timeout: 0
Time: 2024-09-11T09:06:55Z, Total Conns: 500, Idle Conns: 500, Stale Conns: 7556, Hits: 667712, Misses: 37998, Timeout: 0
Time: 2024-09-11T09:07:55Z, Total Conns: 500, Idle Conns: 500, Stale Conns: 7556, Hits: 667761, Misses: 37998, Timeout: 0

```

Since i kept the IdleTimeout for 2 minutes, shouldn't these connections be closed after that and my count be 0. Bt i couldn't see any stale connections decreasing but keeps on increasing. 

Below is the code iam using to print these stats:

`stats := redisClient.PoolStats() `
`logString := fmt.Sprintf("Time: %s, Total Conns: %d, Idle Conns: %d, Stale Conns: %d, Hits: %d, Misses: %d, Timeout: %d\n",
			time.Now().Format(time.RFC3339), stats.TotalConns, stats.IdleConns, stats.StaleConns, stats.Hits, stats.Misses, stats.Timeouts)
		`

**Comment by recodyx** [comment]
It is a normal behavior. 

stale connection metric describes "how many connection are removed from the pool". This is a commulative metric and will not reset.

https://pkg.go.dev/github.com/go-redis/redis@v6.15.9+incompatible/internal/pool#Stats
